src/main/java/org/primefaces/component/terminal/TerminalRenderer.java

```diff
@@ -102,17 +110,40 @@ protected void encodeScript(FacesContext context, Terminal terminal) throws IOEx
     }
     
     protected void handleCommand(FacesContext context, Terminal terminal) throws IOException {
-        ResponseWriter writer = context.getResponseWriter();
-        String clientId = terminal.getClientId(context);
-        String value = context.getExternalContext().getRequestParameterMap().get(clientId + "_input");
-        String tokens[] = value.split(" ");
+        String tokens[] = getValueTokens(context, terminal);
         String command = tokens[0];
         String[] args = Arrays.copyOfRange(tokens, 1, tokens.length);
 
         MethodExpression commandHandler = terminal.getCommandHandler();
         String result = (String) commandHandler.invoke(context.getELContext(), new Object[]{command, args});
 
+        ResponseWriter writer = context.getResponseWriter();
         writer.write(result);
     }
 
+    protected void autoCompleteCommand(FacesContext context, Terminal terminal) throws IOException {
+        String tokens[] = getValueTokens(context, terminal);
+        String command = tokens[0];
+        String[] args = Arrays.copyOfRange(tokens, 1, tokens.length);
+
+        TerminalAutoCompleteModel autoCompleteModel = terminal.getAutoCompleteModel();
+        ResponseWriter writer = context.getResponseWriter();
+        if (autoCompleteModel == null) {
+            writer.write("null");
+        }
+        else {
+            TerminalAutoCompleteMatches matches = terminal.traverseAutoCompleteModel(autoCompleteModel, command, args);
+            writer.write(matches.toString());
+        }
+    }
+
+    private String[] getValueTokens(FacesContext context, Terminal terminal) {
+        String clientId = terminal.getClientId(context);
+        String value = context.getExternalContext().getRequestParameterMap().get(clientId + "_input");
```

Comment:
My only thought here is that OWASP says that all RequestParameters should be treated as hostile in case someone injects a request param in the map.  Maybe you should pass the value though ComponentUtils.escapeText() function or something similar?   

https://www.owasp.org/index.php/Java_Server_Faces#ManagedBeans

As a side note for myself maybe we need to think about this as a whole across the PrimeFaces codebase when accepting Request Parameters.

Reply:
While working in this area I assumed this is ok as the `handleCommand` part was already doing the same (just introduced `getValueTokens()` to reuse logic), so in order to stay consistent I did not change that. Also I have found other components that do processing the same way, therefore identified this as a more global issue. I can change this of course if project-wise there is already a convention about how to deal with this. I could not find that in my brief research.

Reply:
No don't make any changes for that I was just pointing it out. I think whatever PF comes up with we will need to look at every single get() value in the framework.  So no changes needed right now!